docs/data-sources/confluent_provider_integration_setup.md

```diff
@@ -0,0 +1,92 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "confluent_provider_integration_setup Data Source - terraform-provider-confluent"
+subcategory: ""
+description: |-
+  
+---
+
+# confluent_provider_integration_setup Data Source
+
+[![General Availability](https://img.shields.io/badge/Lifecycle%20Stage-General%20Availability-%2345c6e8)](https://docs.confluent.io/cloud/current/api.html#section/Versioning/API-Lifecycle-Policy)
+
+`confluent_provider_integration_setup` describes a Cloud Service Provider (CSP) integration that allows Confluent Cloud to access resources in your cloud provider account.
+
+## Example Usage
+
+### Azure Provider Integration
+
+```terraform
+data "confluent_provider_integration_setup" "azure" {
+  id = "cspi-abc123"
+  environment {
+    id = "env-xyz456"
+  }
+}
+
+output "azure_integration_status" {
+  value = data.confluent_provider_integration_setup.azure.status
+}
+```
+
+### GCP Provider Integration
+
+```terraform
+data "confluent_provider_integration_setup" "gcp" {
+  display_name = "my-gcp-integration"
+  environment {
+    id = "env-xyz456"
+  }
+}
+
+output "gcp_integration_id" {
+  value = data.confluent_provider_integration_setup.gcp.id
+}
+```
+
+### Using with Authorization Resource
+
+```terraform
+data "confluent_provider_integration_setup" "main" {
+  id = "cspi-abc123"
+  environment {
+    id = "env-xyz456"
+  }
+}
+
+# Use the integration with authorization data source
+data "confluent_provider_integration_authorization" "main" {
+  id = data.confluent_provider_integration_setup.main.id
+  environment {
+    id = "env-xyz456"
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Argument Reference
+
+The following arguments are supported:
+
+- `environment` - (Required Configuration Block) supports the following:
+    - `id` - (Required String) The ID of the Environment that the Provider Integration belongs to, for example, `env-abc123`.
+- `id` - (Optional String) The ID of the Provider Integration, for example, `cspi-4xg0q`. Exactly one of `id` or `display_name` must be specified.
+- `display_name` - (Optional String) The display name of the Provider Integration. Exactly one of `id` or `display_name` must be specified.
+
+## Attributes Reference
+
+In addition to the preceding arguments, the following attributes are exported:
+
+- `id` - (Required String) The ID of the Provider Integration, for example, `cspi-4xg0q`.
+- `display_name` - (Required String) The display name of the Provider Integration.
+- `cloud_provider` - (Required String) The cloud service provider. Values are `azure` and `gcp`.
```

Comment:
```suggestion
- `cloud` - (Required String) The cloud service provider. Values are `AZURE` and `GCP`.
```

See https://github.com/confluentinc/terraform-provider-confluent/blob/master/internal/provider/resource_network.go#L76 for more details.